client/telemetry/TelemetryEventClient.go

```diff
@@ -0,0 +1,321 @@
+		return
+	}
+	prop := make(map[string]interface{})
+	err = json.Unmarshal(reqBody, &prop)
+	if err != nil {
+		impl.logger.Errorw("SummeryEventForTelemetry, payload unmarshal error", "error", err)
+		return
+	}
+	impl.PosthogClient.Client.Enqueue(posthog.Capture{
+		DistinctId: ucid,
+		Event:      fmt.Sprintf("summary event sent from devtron"),
+		Properties: prop,
+	})
+}
+
+func (impl *TelemetryEventClientImpl) HeartbeatEventForTelemetry() {
+	clusterBean, err := impl.clusterService.FindOne(cluster.ClusterName)
+	if err != nil {
+		impl.logger.Errorw("exception caught inside telemetry heartbeat event", "err", err)
+		return
+	}
+	cfg, err := impl.clusterService.GetClusterConfig(clusterBean)
+	if err != nil {
+		impl.logger.Errorw("exception caught inside telemetry heartbeat event", "err", err)
+		return
+	}
+
+	client, err := impl.K8sUtil.GetClientForIncluster(cfg)
+	if err != nil {
+		impl.logger.Errorw("exception caught inside telemetry heartbeat event", "err", err)
+		return
+	}
+	cm, err := impl.K8sUtil.GetConfigMap(impl.aCDAuthConfig.ACDConfigMapNamespace, DevtronUniqueClientIdConfigMap, client)
+	if err != nil && strings.Contains(err.Error(), "not found") {
+		// if not found, create new cm
+		//cm = &v12.ConfigMap{ObjectMeta: v13.ObjectMeta{Name: "devtron-upid"}}
+		cm = &v1.ConfigMap{ObjectMeta: v12.ObjectMeta{Name: DevtronUniqueClientIdConfigMap}}
+		data := map[string]string{}
+		data[DevtronUniqueClientIdConfigMapKey] = util.Generate(16) // generate unique random number
+		cm.Data = data
+		_, err = impl.K8sUtil.CreateConfigMap(impl.aCDAuthConfig.ACDConfigMapNamespace, cm, client)
+		if err != nil {
+			impl.logger.Errorw("exception caught inside telemetry heartbeat event", "err", err)
+			return
+		}
+	}
+	if cm == nil {
+		impl.logger.Errorw("configmap found nil for telemetry heartbeat event", "cm", cm)
+		return
+	}
+	dataMap := cm.Data
+	ucid := dataMap[DevtronUniqueClientIdConfigMapKey]
+
+	discoveryClient, err := impl.K8sUtil.GetK8sDiscoveryClient(cfg)
+	if err != nil {
+		impl.logger.Errorw("exception caught inside telemetry heartbeat event", "err", err)
+		return
+	}
+	k8sServerVersion, err := discoveryClient.ServerVersion()
+	if err != nil {
+		impl.logger.Errorw("exception caught inside telemetry heartbeat event", "err", err)
+		return
+	}
+	payload := &TelemetryEventDto{UCID: ucid, Timestamp: time.Now(), EventType: Heartbeat, DevtronVersion: "v1"}
+	payload.ServerVersion = k8sServerVersion.String()
+	reqBody, err := json.Marshal(payload)
+	if err != nil {
+		impl.logger.Errorw("HeartbeatEventForTelemetry, payload marshal error", "error", err)
+		return
+	}
+	prop := make(map[string]interface{})
+	err = json.Unmarshal(reqBody, &prop)
+	if err != nil {
+		impl.logger.Errorw("HeartbeatEventForTelemetry, payload unmarshal error", "error", err)
+		return
+	}
+	impl.PosthogClient.Client.Enqueue(posthog.Capture{
+		DistinctId: ucid,
+		Event:      fmt.Sprintf("heartbeat event sent from devtron"),
```

Comment:
use short keywords as event name, filtering will be easier

Reply:
done